src/SoftRobots.Inverse/component/constraint/BarycentricCenterEffector.inl

```diff
@@ -77,31 +76,27 @@ BarycentricCenterEffector<DataTypes>::~BarycentricCenterEffector()
 template<class DataTypes>
 void BarycentricCenterEffector<DataTypes>::init()
 {
-    Inherit1::init();
-
-    if(m_state==nullptr)
-        msg_error() << "There is no mechanical state associated with this node. "
-                       "the object is deactivated. "
-                       "To remove this error message fix your scene possibly by "
-                       "adding a MechanicalObject." ;
-    initData();
-
-    d_drawBarycenter.setGroup("Visualization");
-    d_barycenter.setReadOnly(true);
+    // d_indices is inherited from PositionEffector.
+    // Even though we do not use this data we need to initilize it to avoid warnings
+    auto indices = sofa::helper::getWriteAccessor(d_indices);
+    indices.resize(1);
 
-    computeBarycenter();
-}
+    Inherit1::init();
 
-template<class DataTypes>
-void BarycentricCenterEffector<DataTypes>::reinit()
-{
-    initData();
+    if (d_axis.isSet())
+    {
+        msg_deprecated() << "The data axis is deprecated. To fix your scene please use useDirections instead. It will be remove in v25.06.";
+        auto useDirections = sofa::helper::getWriteAccessor(d_useDirections);
+        const auto& axis = sofa::helper::getReadAccessor(d_axis);
+        useDirections[0] = axis[0];
+        useDirections[1] = axis[1];
+        useDirections[2] = axis[2];
```

Comment:
```suggestion
        useDirections = axis;
```
to be tested

Reply:
useDirections is a Vec6 since the component handles Rigid3 now, axis is a Vec3